integrations/messenger/src/utils/oauth.ts

```diff
@@ -0,0 +1,77 @@
+import type { IntegrationContext, Request, Response } from '@botpress/sdk'
+import axios from 'axios'
+import { MessengerClient } from 'messaging-api-messenger'
+import z from 'zod'
+import * as bp from '.botpress'
+
+export class MessengerOauthClient {
+  private clientId: string
+  private clientSecret: string
+
+  constructor() {
+    this.clientId = bp.secrets.APP_ID
+    this.clientSecret = bp.secrets.APP_SECRET
+  }
+
+  async getAccessToken(code: string) {
+    const query = new URLSearchParams({
+      client_id: this.clientId,
+      client_secret: this.clientSecret,
+      redirect_uri: `${process.env.BP_WEBHOOK_URL}/oauth`,
+      code,
+    })
+
+    const res = await axios.get(`https://graph.facebook.com/v18.0/oauth/access_token?${query.toString()}`)
+
+    const data = z
+      .object({
+        access_token: z.string(),
+      })
+      .parse(res.data)
+
+    return data.access_token
+  }
+}
+
+export async function handleOAuthRedirect(req: Request, client: bp.Client, ctx: IntegrationContext): Promise<Response> {
+  const slackOAuthClient = new MessengerOauthClient()
```

Comment:
```suggestion
  const messengerOAuthClient = new MessengerOauthClient()
```

Reply:
Changed in https://github.com/botpress/botpress/pull/12912/commits/6078a4520f4897e941225d3ffb46733634200916